modules/RsIskraMeter/src/main.rs

```diff
@@ -0,0 +1,1034 @@
+//! Module for Iskra's WM3M4 & WM3M4C three-phase electrical energy meters.
+//!
+//! The implementation follows the user manual
+//! <https://www.iskra.eu/f/docs/Smart-energy-meters/K_WM3M4_EN_22433922_Users_manual_Ver_1.14.pdf>
+//! 
+//! ## Example usage:
+//! 
+//! To try out the Iskra meter you can use the following dummy config below.
+//! 
+//! ```yaml
+//!active_modules:
+//!  power_meter_1:
+//!    module: RsIskraMeter
+//!    config_module:
+//!      powermeter_device_id: 33
+//!    connections:
+//!      modbus:
+//!      - module_id: serial_comm_hub_1
+//!        implementation_id: main
+//!  serial_comm_hub_1:
+//!    module: SerialCommHub
+//!    config_implementation:
+//!      main:
+//!        serial_port: /dev/ttyUSB0
+//!        baudrate: 115200
+//!        initial_timeout_ms: 10000
+//!        within_message_timeout_ms: 100
+//!        max_packet_size: 100
+//! ```
+//! 
+//! You can start and stop transactions by publishing following messages to
+//! mqtt
+//! 
+//! ```sh
+//! mosquitto_pub -t everest/power_meter_1/meter/cmd -m '{"data":{"args":{"value":{"cable_id":1,"client_id":"","evse_id":"DEABCD312ABC11","tariff_id":1,"transaction_id":"foobarbaz","user_data":""}},"id":"foo","origin":"bar"},"name":"start_transaction","type":"call"}'
+//! mosquitto_pub -t everest/power_meter_1/meter/cmd -m '{"data":{"args":{"transaction_id":"foobarbaz"}, "id":"foo","origin":"bar"},"name":"stop_transaction","type":"call"}'
+//! ```
+#![allow(non_snake_case, non_camel_case_types)]
+
+include!(concat!(env!("OUT_DIR"), "/generated.rs"));
+mod utils;
+mod logger;
+
+use anyhow::Result;
+use chrono::{Local, Offset, Utc};
+use generated::types::powermeter::{
+    Powermeter, TransactionRequestStatus, TransactionStartResponse, TransactionStopResponse,
+};
+use generated::types::serial_comm_hub_requests::{StatusCodeEnum, VectorUint16};
+use generated::types::units::{Current, Energy, Frequency, Power, ReactivePower, Voltage};
+use generated::types::units_signed::SignedMeterValue;
+use generated::{get_config, Module, SerialCommunicationHubClientPublisher};
+use std::fmt::Debug;
+use std::sync::{Arc, Mutex};
+use std::time::Duration;
+use utils::{
+    counter, create_ocmf, create_random_meter_session_id, from_t5_format, from_t6_format,
+    string_to_vec, to_8_string, to_hex_string,
+};
+
+/// Public key prefix for transparency software, defined under 6.5.14.
```

Comment:
Could these definitions and conversions be moved into another file for readability?

Reply:
below most stuff is derived from the iskra manual linked at the top of the file... splitting it up would probably cause readers to flip back and forth between the enums and the impl - but I don't have a strong opinion here

 @hikinggrass  how much does it bother you ?

Reply:
Ok that makes sense, it's not a dealbreaker :wink: